IPython/core/application.py

```diff
@@ -294,21 +303,22 @@ def _ipython_dir_changed(self, change):
             str_old = os.path.abspath(old)
             if str_old in sys.path:
                 sys.path.remove(str_old)
-        str_path = os.path.abspath(new)
-        sys.path.append(str_path)
-        ensure_dir_exists(new)
-        readme = os.path.join(new, 'README')
-        readme_src = os.path.join(get_ipython_package_dir(), u'config', u'profile', 'README')
-        if not os.path.exists(readme) and os.path.exists(readme_src):
-            shutil.copy(readme_src, readme)
-        for d in ('extensions', 'nbextensions'):
-            path = os.path.join(new, d)
-            try:
-                ensure_dir_exists(path)
-            except OSError as e:
-                # this will not be EEXIST
-                self.log.error("couldn't create path %s: %s", path, e)
-        self.log.debug("IPYTHONDIR set to: %s" % new)
+        if self.add_ipython_dir_to_sys_path:
+            str_path = os.path.abspath(new)
+            #sys.path.append(str_path)
```

Comment:
I'm not sure this is right - the append is unconditionally skipped, and the subdir creation presumably shouldn't be under this condition?

Reply:
```suggestion
            sys.path.append(str_path)
```

Reply:
Yeah, like a typo just before committing. 

I"d like to remove this option in the future, so my hope was to never create the dir if it is not needed.
Nowadays people can install extensions with pip so nothing should go in that dir maybe ?

I need to try with the cython extension see how it behaves.